src/components/override-sdk/field/AutoComplete/AutoComplete.tsx

```diff
@@ -159,23 +159,50 @@ export default function AutoComplete(props: AutoCompleteProps) {
     }
   }, [theDatasource]);
 
+  const [currentLang, setCurrentLang] = useState(
+    sessionStorage.getItem('rsdk_locale')?.substring(0, 2).toUpperCase() || 'EN'
+  );
+
+  PCore.getPubSubUtils().subscribe(
+    'languageToggleTriggered',
+    ({ language }) => {
+      setCurrentLang(language.toUpperCase());
+    },
+    ''
+  );
+
   useEffect(() => {
+    if (currentLang === 'CY') {
+      PCore.getLocaleUtils().loadLocaleResources([
+        `@BASECLASS!DATAPAGE!${datasource.toUpperCase()}`
+      ]);
+    }
+
     if (!displayMode && listType !== 'associated') {
       getDataPage(datasource, parameters, context).then((results: any) => {
         const optionsData: Array<any> = [];
         const displayColumn = getDisplayFieldsMetaData(columns);
+        const translationDataPage = `@BASECLASS!DATAPAGE!${datasource.toUpperCase()}`;
+        const localePath = datasource === 'D_NationalityList' ? 'Value' : 'CountryName';
```

Comment:
is there a scope to get this datasource dynamically from Pega props directly?

Reply:
I'm not sure which line you are referring to. 185 or 186? Or exactly what you mean. Do you mean the actual 'datasource' which is coming from the props. Or something else. Let me know if a call would be easier.